### ISSUE TITLE
Road to 1.0.0

### ISSUE THREAD

**Issue by seanpianka** [logannc/fuzzywuzzy-rs#21]
I've added a Reddit [post here](https://www.reddit.com/r/rust/comments/jmby4s/fuzzywuzzyrs_a_port_of_the_fuzzywuzzy_python/) to try and give visibility to the crate. Hopefully Google will recommend the crate in any "rust fuzzy string matching" query!

@logannc, I think we should decide on the public API to go forward with, w.r.t. your thoughts [here](https://github.com/logannc/fuzzywuzzy-rs/pull/15#issuecomment-699563273). We should stabilize on a public API and properly version it at 1.0.0 and beyond.

Thinking about the changes that must made before bumping to 1.0.0, I see this list remains:
* https://github.com/logannc/fuzzywuzzy-rs/pull/15 - Explicit type definitions for matches and scoring (or is this not necessary?)
* https://github.com/logannc/fuzzywuzzy-rs/issues/19 - String alignment algorithm
* #20 - Unicode support
* #7 - Missing methods from the original implementation of the `process` module
* #6 - Only run processor methods once in the `extract_*` function implementations

I think once we reach feature parity with the original fuzzywuzzy, we can bump the version. Thoughts?

**Comment by logannc** [comment]
I agree. Just been busy with other things. I appreciate you helping organize this.

#15 is probably required because it would probably be a breaking change
#19 is an new feature so while I'm more excited about it than code cleanup, it should not block 1.0
#20 would be ideal to figure out before 1.0 because it could change results and/or apis
#7 yup
#6 like #19, it'd be nice to have but is a performance improvement and as long as there are not changes to the results, this can happen any time.

**Comment by logannc** [comment]
I will also note that I am, unfortunately, quite busy for the next few weeks.
